### PR TITLE
Complete login use case

### LINKED ISSUE #14: Complete login use case
Now that we're done the team use case, complete the login use case/related views after logging in:

- [x] Update the Java Swing code for a UI that is consistent with the signup view
- [x] The user can cancel from the login view, going back to signup view
- [x] Show appropriate popups if user does not exist or if password is incorrect
- [x] Add appropriate unit tests
- [x] Add appropriate Javadocs

### PR DESCRIPTION
**Pull request by az19642**
Closes #14
In addition to the tasks specified in issue, LoginView no longer implements ActionListener and only using anonymous listeners. This aligns better with SRP.

### CODE REVIEW

File: src/main/java/app/LoginUseCaseFactory.java [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import interface_adapter.login.LoginController;
 import interface_adapter.login.LoginPresenter;
 import interface_adapter.login.LoginViewModel;
+import interface_adapter.signup.SignupViewModel;
```
Comment (jchrisoh) [inline src/main/java/app/LoginUseCaseFactory.java:10]: Good - import statements do not violate dependency rule

File: src/main/java/interface_adapter/login/LoginPresenter.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ public void prepareSuccessView(LoginOutputData response) {
     @Override
     public void prepareFailView(String error) {
```
Comment (jchrisoh) [inline src/main/java/interface_adapter/login/LoginPresenter.java:50]: Write Javadocs for public methods

Comment (jchrisoh) [inline src/main/java/interface_adapter/login/LoginPresenter.java:50]: Write Javadocs for public methods

File: src/main/java/interface_adapter/login/LoginState.java [hunk 3]
```diff
@@ -91,4 +93,12 @@ public String getPasswordError() {
     public void setPasswordError(String passwordError) {
         this.passwordError = passwordError;
     }
+
+    public void setLoginError(String loginError) {
```
Comment (jchrisoh) [inline src/main/java/interface_adapter/login/LoginState.java:97]: Write Javadocs for public methods

File: src/main/java/view/LoginView.java [hunk 4]
```diff
@@ -17,54 +17,59 @@
...
+    private final JButton cancel;
     private final LoginViewModel loginViewModel;
+    private final SignupViewModel signupViewModel;
+    private final ViewManagerModel viewManagerModel;
     private final JLabel usernameErrorField = new JLabel();
     private final JLabel passwordErrorField = new JLabel();
     private final LoginController loginController;
 
-    public LoginView(LoginViewModel loginViewModel, LoginController controller) {
+    public LoginView(LoginViewModel loginViewModel, SignupViewModel signupViewModel, LoginController controller,
```
Comment (jchrisoh) [inline src/main/java/view/LoginView.java:34]: Nice use of dependency injection strategy design

**Comment by jchrisoh** [comment]
I will accept the pull request once Javadocs have been written for all public methods

File: src/main/java/interface_adapter/login/LoginPresenter.java (hunk 2 above)
Comment (az19642) [reply src/main/java/interface_adapter/login/LoginPresenter.java:50]: There is a Javadoc for this method.

File: src/main/java/view/LoginView.java [hunk 5]
```diff
@@ -17,54 +17,59 @@
...
  */
-public class LoginView extends JPanel implements ActionListener, PropertyChangeListener {
+public class LoginView extends JPanel implements PropertyChangeListener {
 
     public final String viewName = "log in";
-    final JTextField usernameInputField = new JTextField(15);
-    final JPasswordField passwordInputField = new JPasswordField(15);
-    final JButton logIn;
-    final JButton cancel;
+    private final JTextField usernameInputField = new JTextField(15);
```
Comment (jchrisoh) [inline src/main/java/view/LoginView.java:23]: Good choice of access modifiers

**Comment by jchrisoh** [comment]
Login use case is complete, and Javadocs have been written as suggested during review. The implementation follows SRP and DIP and also adheres to Clean Architecture. This issue can be closed.

**Coding Style**: Appropriate naming conventions and concise button labels in LoginView
**Correctness**: Was able to check out successfully and run the code. Commit messages describing use case are all correct.
**SOLID + CA**: LoginView action listener adheres to SRP and dependency injection is used in many places like LoginView. Packaging structure follows Clean Architecture and no new files were needed to be made.
**Javadocs + Tests**: Good; all public methods and classes have been covered after review. Tests run successfully and coverage is 100%.
**Clarity**: Clear and descriptive commit messages and pull request description.

**Review by jchrisoh — APPROVED** [review]
Login use case is complete, and Javadocs have been written as suggested during review. The implementation follows SRP and DIP and also adheres to Clean Architecture. This issue can be closed.

**Coding Style**: Appropriate naming conventions and concise button labels in LoginView
**Correctness**: Was able to check out successfully and run the code. Commit messages describing use case are all correct.
**SOLID + CA**: LoginView action listener adheres to SRP and dependency injection is used in many places like LoginView. Packaging structure follows Clean Architecture and no new files were needed to be made.
**Javadocs + Tests**: Good; all public methods and classes have been covered after review. Tests run successfully and coverage is 100%.
**Clarity**: Clear and descriptive commit messages and pull request description.

**Pull request merged**